### DISCUSSION TITLE
Dynamic import from CDN

### DISCUSSION THREAD

**Discussion by sabativi** [meteor/meteor#11798] [Feature Requests]
Hello, it is easy to setup a CDN for meteor, but when doing dynamic import, all the bundles are fetch from the server which can cause massive scalability problem.

**Comment by paulincai** [comment]
Hi @sabativi,

I looked into this a while back and the problem in general is the dynamic packages/mini-bundles do not have specific names like the Meteor bundle has.
The way you get them is via a POST message. So far I studied Cloudfront and basically you need to ask Cloudfront for a certain package name and if that is not in the CDN , Cloudfront is supposed to send a POST to Meteor, get it and serve it. The client needs to know what the package name should be.

I think the changes required are pretty extensive. Luckily these mini-bundles get cached in the browsers and in theory a user will ask for it browsers * Meteor bundle version times.

Are you (I mean a project of yours :) struggling in any way?

**Reply by sabativi** [reply]
I am using dynamic import quite intensively in my app as it is a big app with different roles.
I want do not wand to overload galaxy servers with bundles fetching.
